### PR TITLE
Feat: add ObservedGeneration to GameServerSet

### PR DESCRIPTION
**Pull request by Somefive**
Add ObservedGeneration to GameServerSet.

1. Update api types and crd.
2. Add status update logic.

### CODE REVIEW

File: apis/v1alpha1/gameserverset_types.go [hunk 1]
```diff
@@ -139,6 +139,9 @@ type GameServerSetStatus struct {
 	WaitToBeDeletedReplicas *int32 `json:"waitToBeDeletedReplicas,omitempty"`
 	// LabelSelector is label selectors for query over pods that should match the replica count used by HPA.
 	LabelSelector string `json:"labelSelector,omitempty"`
+	// The generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```
Comment (zmberg) [inline apis/v1alpha1/gameserverset_types.go:144]: Please move the ObservedGeneration to 131 line.

File: pkg/controllers/gameserverset/gameserverset_manager.go [hunk 2]
```diff
@@ -364,6 +366,7 @@ func (manager *GameServerSetManager) SyncStatus() error {
 		MaintainingReplicas:     pointer.Int32Ptr(int32(maintainingGs)),
 		WaitToBeDeletedReplicas: pointer.Int32Ptr(int32(waitToBeDeletedGs)),
 		LabelSelector:           asts.Status.LabelSelector,
+		ObservedGeneration:      gss.GetGeneration(),
 	}
```
Comment (zmberg) [inline pkg/controllers/gameserverset/gameserverset_manager.go:371]: Please whether the old and new status is equal, If no changed, do not need to patch.

File: apis/v1alpha1/gameserverset_types.go (hunk 1 above)
Comment (Somefive) [reply apis/v1alpha1/gameserverset_types.go]: Done

**Pull request merged**